### PR TITLE
Update and clarify JWT docs

### PR DESCRIPTION
**Pull request by ayoho**
Add some sections on JWK usage in the JWT builder and consumer doc.

For https://github.com/OpenLiberty/open-liberty/issues/26453

### CODE REVIEW

**Review by jimmy1wu — APPROVED** [review]

File: modules/reference/pages/feature/jwt/examples.adoc [hunk 1]
```diff
@@ -16,6 +16,39 @@ The `issuer` attribute in the example is the `\http://example.com` URL that iden
...
+. Dynamically generate a JWK.
+. Generate a JWK based on a key in the keystore that is configured for the builder.
+
+In both cases, the JWT builder uses the key represented by the JWK to sign tokens that it creates.
+
+===== Dynamic JWK generation
+
+JWT builders will dynamically generate JWKs to use when the `jwkEnabled` attribute for the `jwtBuilder` element is set to `true`. A new key is generated when the JWT builder creates its first token, or when the JWK endpoint for the builder is invoked, whichever happens first. After creation, the key is rotated automatically every 12 hours.
+
+Dynamically generated JWKs are useful for server administrators that do not want to be concerned with setting up a keystore and signing key for the JWT builder. Server admins need only set the `jwkEnabled` attribute to `true`, and Liberty will handle creation and maintenance of the key. Note that when `jwkEnabled` is set to `true`, the `keyStoreRef` and `keyAlias` attributes will be ignored.
```
Comment (jimmy1wu) [inline modules/reference/pages/feature/jwt/examples.adoc]:
```suggestion
Dynamically generated JWKs are useful for server administrators that do not want to be concerned with setting up a keystore and signing key for the JWT builder. Server administrators only need to set the `jwkEnabled` attribute to `true`, and Liberty will handle creation and maintenance of the key. Note that when `jwkEnabled` is set to `true`, the `keyStoreRef` and `keyAlias` attributes will be ignored.
```

File: modules/reference/pages/feature/jwt/examples.adoc [hunk 2]
```diff
@@ -16,6 +16,39 @@ The `issuer` attribute in the example is the `\http://example.com` URL that iden
...
+
+To use static JWKs, server administrators must configure the `keyStoreRef` and `keyAlias` attributes in the `jwtBuilder` element to point to a specific key to sign tokens. The `jwkEnabled` attribute must be set to `false`, but the JWK endpoint for the JWT builder will still be active. In this case, the JWK endpoint provided by the JWT builder will produce a JSON Web Key Set (JWKS) document that includes the key referenced by the `keyStoreRef` and `keyAlias` attributes.
+
+==== JWT Consumer
+
+JWT consumers can retrieve JSON Web Key Set (JWKS) documents to reconstruct public keys to use for verifying token signatures. JWKS documents are particularly useful for JWT consumers that receive tokens from issuers that frequently rotate their signing keys, and for server administrators who do not want to manually maintain a keystore.
+
+The `jwkEnabled` attribute in the `jwtConsumer` element must be set to `true`, and the `jwkEndpointUrl` attribute must be set to a URI that provides the JWKS document containing the public keys to use to verify the signatures of tokens received by this JWT consumer.
```
Comment (jimmy1wu) [inline modules/reference/pages/feature/jwt/examples.adoc]:
```suggestion
The `jwkEnabled` attribute in the `jwtConsumer` element must be set to `true`, and the `jwkEndpointUrl` attribute must be set to a URI that provides the JWKS document containing the public keys to use to verify the signatures of the tokens received by this JWT consumer.
```

**Comment by dmuelle** [comment]
opened #6964 to track on docs side

**Review by dmuelle — CHANGES_REQUESTED** [review]
Ho @ayoho - thanks for opening this. Some general comments before I review/edit

The Examples sections on the feature pages are intended to be for concise configuration examples that demonstrate common tasks. The info in this PR seems more like conceptual information to support JWT Builder. Is it possible to revise this information to be focused on config examples that demonstrate the main ideas? It might also make the sections more concise if we show rather than tell the config details, and just provide context in the example descriptions.

Also, the OL website does not really recognize h4 or h5 headings. But if the info was broken up into examples, we could just use h3 for each example.

I'm happy to help revise the info you have here. If the information is better presented as conceptual topic, we can create a new topic and link it form this page. Let me know what you think. Thanks!

**Comment by ayoho** [comment]
Hi, @dmuelle. I'd agree that the content is primarily conceptual. Perhaps revamping the https://openliberty.io/docs/latest/single-sign-on.html#_json_web_token_jwt section and maybe even having a standalone page under the Security section for JSON Web Tokens would be a better place to have this information. I can create a new `modules/ROOT/pages/json-web-token.adoc` page for this content, but if we go that route I might need some more help with how to get it correctly hooked into the site. Looks like I'd update `modules/ROOT/nav.adoc` to add a link to the new page where I want under the Security section. Anything else?

**Comment by dmuelle** [comment]
that's all you need to do- BUT - when you update the nav, make that change in the `draft-nav` branch and create a separate PR to `draft`.  We have dedicated branches just for updating the draft and staging nav files to avoid merge conflicts since the nav is different on draft vs staging/vNext.

**Comment by dmuelle** [comment]
https://github.com/OpenLiberty/docs#editing-the-docs-navigation

**Comment by ayoho** [comment]
@dmuelle I moved all my new content into a separate json-web-token.adoc page. I'll open a separate PR for the nav changes.

**Review by dmuelle — APPROVED** [review]
LGTM

**Comment by dmuelle** [comment]
I'm merging this to draft and will review further and make edits for IBM style etc in a new PR

**Pull request merged**